src/config.rs

```diff
@@ -414,6 +471,101 @@ impl Config {
     pub fn mirror_map(&self) -> &std::collections::HashMap<Url, Vec<Url>> {
         &self.mirrors
     }
+
+    /// Modify this config with the given key and value
+    ///
+    /// # Note
+    ///
+    /// It is required to call `save()` to persist the changes.
+    pub fn set(&mut self, key: &str, value: Option<String>) -> miette::Result<()> {
+        match key {
+            "default-channels" => {
+                self.default_channels = value
+                    .map(|v| serde_json::de::from_str(&v))
+                    .transpose()
+                    .into_diagnostic()?
+                    .unwrap_or_default();
+            }
+            "change-ps1" => {
+                self.change_ps1 = value.map(|v| v.parse()).transpose().into_diagnostic()?;
+            }
+            "tls-no-verify" => {
+                self.tls_no_verify = value.map(|v| v.parse()).transpose().into_diagnostic()?;
+            }
+            "repodata-config.disable-jlap" => {
+                self.repodata_config
+                    .get_or_insert(RepodataConfig::default())
+                    .disable_jlap = value.map(|v| v.parse()).transpose().into_diagnostic()?;
+            }
+            "repodata-config.disable-bzip2" => {
```

Comment:
Would it be possible to split this code on `.`?

Reply:
I don't get this, could you explain?

Reply:
This list of fields will grow. If we could split the code into the different tables it might be easier to maintain.

Reply:
I've refactored the pattern matching to split on `.`, everything still hardcoded in the set method though. Please let me know if you have any better way to achieve.